airbyte_cdk/sources/streams/concurrent/cursor.py

```diff
@@ -442,3 +446,19 @@ def _evaluate_upper_safely(self, lower: CursorValueType, step: GapType) -> Curso
             return lower + step
         except OverflowError:
             return self._end_provider()
+
+    def should_be_synced(self, record: Record) -> bool:
+        """
+        Determines if a record should be synced based on its cursor value.
+        :param record: The record to evaluate
+
+        :return: True if the record's cursor value falls within the sync boundaries
+        """
+        try:
+            record_cursor_value: CursorValueType = self._extract_cursor_value(record)  # type: ignore  # cursor_field is converted to an InterpolatedString in __post_init__
+        except ValueError:
+            LOGGER.warning(
```

Comment:
I was checking source-pipedrive for the migration to concurrency and this log might be very noisy. Can we only log once for that? Note that we will have the same issue in `observe` so we might want to add the try/except here too @darynaishchenko

Reply:
We can start once per stream for now and see if we need more eventually

Reply:
added